src/lib/scoringUtils.ts

```
@@ -1,324 +1,373 @@
+      const valueToNorm = component.valueUsedForNormalization; // This is now either the original valid number or an imputed valid number
+      const higherIsBetter = metricConfig.higherIsBetter; // Removed !== false, as boolean is direct.
+
+      component.normalizationMethod = normalizationMode;
+      component.log.push(`Normalizing: ${valueToNorm?.toFixed(3)}. Mode: ${normalizationMode}. HigherIsBetter: ${higherIsBetter}`);
+
+      try {
+        if (component.hasUsableValue) { // valueToNorm is guaranteed to be a finite number here
+          const finiteValueToNorm = valueToNorm as number;
+          switch (normalizationMode) {
+            case 'global_min_max':
+              const gRange = globalFullRanges[db_column];
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)